packages/keyv/src/index.js

```diff
@@ -73,7 +73,8 @@ class Keyv extends EventEmitter {
 		// Attach iterators
 		if (typeof this.opts.store[Symbol.iterator] === 'function' && this.opts.store instanceof Map) {
 			this.iterator = generateIterator(this.opts.store);
-		} else if (typeof this.opts.store.iterator === 'function' && this.opts.store.opts && this.opts.store.opts.dialect === 'sqlite') { // Iterator only supported for sqlite for now
+		} else if (typeof this.opts.store.iterator === 'function' && this.opts.store.opts
```

Comment:
@alphmth wonder if it would be better for right now to create an array for which ones support it above so you can just add a line each time such as the following:

```javascript
const iterableAdapters = [
'sqlite',
'postgres'
...
]

Then below you can just do a quick check if it is in the array.